### PR TITLE
ENH: adds filename to title and navbar

### PR DESCRIPTION
**Pull request by ebolyen**
Remote file, short name:
image

Local file, long name:
image

### CODE REVIEW

**Comment by ebolyen** [comment]
fixes #42

File: app/ToolbarWindow/Menu.jsx [hunk 1]
```diff
@@ -5,6 +5,16 @@ import queryString from 'query-string';
 
 import logo from './img/q2view.png';
 
+const fileNameStyle = {
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+    width: 'calc(100% - 450px)',
```
Comment (thermokarst) [inline app/ToolbarWindow/Menu.jsx:11]: 👍

File: app/ToolbarWindow/Menu.jsx [hunk 2]
```diff
@@ -45,7 +55,11 @@ export default class Menu extends React.Component {
                     )}
                 </Navbar.Header>
                 {(this.props.hasVisualization || this.props.hasMetadata || this.props.hasProvenance) && ( // eslint-disable-line max-len
-                <Navbar.Collapse>
+                <Navbar.Collapse style={{ textAlign: 'center' }}>
+                    <Navbar.Text style={fileNameStyle} className="hidden-xs">
+                        Current File:
```
Comment (thermokarst) [inline app/ToolbarWindow/Menu.jsx:60]: Personally I think this should be `<blink>Current File:</blink>`, but, to each his own.

File: app/lib/util.js [hunk 3]
```diff
@@ -27,3 +27,7 @@ export const waitUntil = condition => (
         checkCondition();
     })
 );
+
+export const parseFileNameFromURL = urlString => (
+    (new URL(urlString)).pathname.split('/').pop()
```
Comment (thermokarst) [inline app/lib/util.js:32]: ❤️

File: app/init/index.js [hunk 4]
```diff
@@ -119,6 +122,10 @@ export const navigationAction = ({ pathname, query, search, action }) => (dispat
             });
         } else {
             dispatch(loadSuccess());
+            const fileName = getFileName(getState());
+            if (fileName !== null) {
+                document.title = `${fileName} | QIIME 2 View`;
```
Comment (thermokarst) [inline app/init/index.js:127]: Good call on prioritizing the filename in the `<title>`, it is front-and-center in a window with tabs-galore!

**Review by thermokarst — APPROVED** [review]
Sweet! This looks great! Tested locally on Chrome 56 and Firefox 51. Thanks @ebolyen!

**Pull request merged**